tests/APIv2/DiscussionsTest.php

```diff
@@ -289,4 +294,36 @@ public function testEmptyDiscussionTitle() {
         $discussion = $this->api()->get("$this->baseUrl/{$row['discussionID']}")->getBody();
         $this->assertNotEmpty($discussion['name']);
     }
+
+    /**
+     * Announcements should obey the sort.
+     */
+    public function testAnnouncementSort(): void {
+        $this->insertRecords(3, ['Announce' => 1]);
+
+        $fields = ['discussionID', '-discussionID'];
+
+        foreach ($fields as $field) {
+            $rows = $this->api()->get($this->baseUrl, ['pinned' => true, 'sort' => $field])->getBody();
+            $this->assertNotEmpty($rows);
+            $this->assertSorted($rows, $field);
+        }
+    }
+
+    /**
+     * A mix of announcements and discussions should sort properly.
+     */
+    public function testAnnouncementMixed(): void {
+        $rows = $this->insertRecords(2, ['Announce' => 1]);
+        $rows = array_merge($rows, $this->insertRecords(2));
+        $ids = array_column($rows, 'DiscussionID');
+
+        $fields = ['discussionID', '-discussionID'];
+
+        foreach ($fields as $field) {
+            $rows = $this->api()->get($this->baseUrl, ['discussionID' => $ids, 'pinOrder' => 'first', 'sort' => $field])->getBody();
+            $this->assertNotEmpty($rows);
+            $this->assertSorted($rows, '-pinned', $field);
+        }
+    }
```

Comment:
These two tests round off the test coverage of `DiscussionsApiController::index()`.